### DISCUSSION TITLE
Upgrade of Data Rate - RFM98PW SX1278

### DISCUSSION THREAD

**Discussion by tolgaeng** [jgromes/RadioLib#449] [Q&A] [answered]
Hi Jan!

We have a rocket communication system with RFM98PW LoRa modules to send telemetry data packets. But the data rate of the commutation is not satisfied us. We want to get much better results with changes of some parameters. Is it possible? If it is, how?

Here are some details of our system:

As a microcontroller, we have ATMega2560 & STM32F446RE.
The communication system is basically, we have two RFM98PW SX1278 with 433 MHz.
And out range is up to 5 kilometers.

And the results that we get, 100-byte data with communication only takes about 700 ms. 100-byte data with full function (other modules and sensors) takes more than a second.

Is it possible the get about 200 ms with 50-100 bytes of data packets?

Thanks in advance

**Comment by jgromes** [comment, marked as answer]
Short version - use GMSK with PA for this sort of thing, not LoRa.

Long version: In radio communication, you have to balance three basic properties of the system: data rate, range and output power. You can never optimize all three simultaneously - high data rate over long range means you need high output power, to keep sufficient margin in your link budget. If you can't afford high output power and still need the long range capability, you can't transmit fast.

LoRa allows you to trade data rate for improved range while keeping the same output power. Basically, this means that you should optimize your system for low data-rate, infrequent transmissions. That's why it's used for IoT sensors, which can afford to send just a couple bytes once a few minutes, or even hours, and it doesn't matter if the transmission itself takes several seconds.

Your system is the opposite - you (presumably) need to transfer data frequently and with low latency. I suspect rockets tend to fly fast. This means that you can't take advantage of the thing that makes LoRa so powerful: extreme sensitivity at low data rate.

Fortunately, RFM98 (or rather the SX1278 chip that's inside it) can also transmit FSK. If you're careful about your selection of bit rate and frequency deviation, you can transmit MSK (minimum shift keying), which is the most spectrally efficient type of FSK. Add Gaussian shaping to "smooth out" the signal, a strong FEC code and you can improve your link budget quite a bit. A 1k2 - 9k6 baud GMSK with +33 dBm power amplifier is commonly used by CubeSats in LEO (500 - 600 km direct visibility). Granted, the ground segment usually has a large yagi with pointing capability and a much more powerful PA/ sensitive LNA.

For you application, I would use the highest power output you can. Presumably your rocket doesn't fly long, so no need to save power. I would then use a 9.6 kbps bit rate with 1.2 kHz frequency deviation resulting in modulation index of 0.5 - that's MSK. Next, I would apply a Gaussian filter with bandwidth-time product of 0.5. Finally, I would use at least 64-bit preamble, and employ at least Hamming(7, 4) coding, if not Reed-Solomon(255, 223). Then I would test the entire setup, and if something isn't right, I would iterate on those settings.

> 100-byte data with full function (other modules and sensors) takes more than a second.

One final note, if you're sending 100 bytes of data, then that should always take the same amount of time, regardless of whether there's other stuff going on. If this is not the case, then that suggests an issue with your software design.
